src/include/utils.h

```diff
@@ -586,6 +586,16 @@ pgagroal_server_state_as_string(signed char state);
 char*
 pgagroal_connection_state_as_string(signed char state);
 
+/**
+ * Get the Linux kernel version
+ * @param major Pointer to store major version
+ * @param minor Pointer to store minor version
+ * @param patch Pointer to store patch version
+ * @return 0 if success, -1 if error
+ */
+int
+pgagroal_linux_version(int* major, int* minor, int* patch);
```

Comment:
Why not a `pgagroal_os_lernel_version( char* os, int* major, int *minor, int* patch)` and then implement the function so that it can detect also if the system is a BSD one and put it into `os`, filling the other values accordingly?
FreeBSD and OpenBSD us only two digits, with differnt meanings, so `patch` would be always zero or null.